### ISSUE TITLE
to_csv throws `ArgumentError: negative array size` when the limit is greater than the row size.

### ISSUE THREAD

**Issue by sampatbadhe** [ruby/csv#234]
I am adding test cases for recent changes for `limit` option added to `to_csv` method in https://github.com/ruby/csv/pull/229

```ruby
  def test_to_csv_with_limit_set_to_negative_and_greater_than_the_number_of_rows
    csv = <<-CSV
A,B,C
1,2,3
4,5,6
7,8,9
    CSV

    assert_equal(<<-CSV, @table.to_csv(limit: -5))
    CSV
  end

```
When the limit is negative and greater than the row size then `to_csv` throws `ArgumentError: negative array size`

In above case, either it should return all the rows by setting `limit to row size` OR none(just headers when write_headers is true) by setting `limit to zero`.

**Comment by kou** [comment]
@ericgpks Do you want to work on this?

**Comment by sampatbadhe** [comment]
@kou @ericgpks I'd be happy to help here if you could help me with expectations.

**Comment by kou** [comment]
"None" is the expectation.

**Comment by sampatbadhe** [comment]
@kou @ericgpks I have made the changes in https://github.com/ruby/csv/pull/235 please review.

**Comment by ericgpks** [comment]
Thanks a lot. I will check the pr.

**Comment by kou** [comment]
Fixed by #235.
